cartridges/int_klaviyo/cartridge/controllers/KlaviyoRecreate.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+/* API Includes */
+var Logger = require('dw/system/Logger');
+const Resource = require('dw/web/Resource');
+var StringUtils = require('dw/util/StringUtils');
+var URLUtils = require('dw/web/URLUtils');
+
+/* Script Modules */
+var app = require('*/cartridge/scripts/app');
+var guard = require('*/cartridge/scripts/guard');
+var cartModel = require('*/cartridge/scripts/models/CartModel');
+var klaviyoCart = require('*/cartridge/scripts/klaviyo/klaviyoATC');
+var res = require("*/cartridge/scripts/util/Response");
+
+
+/**
+ * KlaviyoRecreate-Cart : The KlaviyoRecreate-Cart route rebuilds a cart page based on a query containing an ENCODED array of objects with product IDs, product quanty, the product's selected options within URL / endpoint.
+ * A DECODED query would be structured as:  <baseURL>/Cart-Recreate?items=[ {"productID": ProdID, "quantity": QTY, "options": [ {lineItemText, optionID, optionValueID} ]} ] *
+ * @param {querystringparameter} - items - JSON containing product Ids, qty, options
+ * @param {renders} - isml via the Cart-Show controller
+ * @param {serverfunction} - Get
+ */
+function cart() {
+    var logger = Logger.getLogger('Klaviyo', 'Klaviyo.SiteGen KlaviyoRecreate.js');
+    try {
+        var cart = app.getModel('Cart').goc();
+        var items = request.httpParameterMap.items ? JSON.parse(StringUtils.decodeBase64(request.httpParameterMap.items)) : null;
+    } catch (error) {
+        logger.error('KlaviyoRecreate-Cart failed. Please check the encoded obj for unexpected chars or syntax issues. ERROR: {0} {1}', error.message, error.stack);
+
+        var additionalErrorContext = Resource.msgf('rebuildcart.message.error.controller.sitegen', 'klaviyo_error', null, error.message);
+        app.getView({
+            message: Resource.msg('rebuildcart.message.error.general.sitegen', 'klaviyo_error', null),
+            errorMessage: Resource.msgf('rebuildcart.message.error.prompt.sitegen', 'klaviyo_error', null, additionalErrorContext),
+        }).render('klaviyo/klaviyoError')
+        return;
+    }
+
+    if (!cart || !items || !items.length) {
+        logger.error(`KlaviyoRecreate-Cart controller failed to create a cart Obj. The currentBasket is ${cart} and items are ${items}.`);
+
+        app.getView({
+            message: Resource.msg('rebuildcart.message.error.general.sitegen', 'klaviyo_error', null),
+            errorMessage: Resource.msgf('rebuildcart.message.error.prompt.sitegen', 'klaviyo_error', null, `The Cart is: ${cart} and Items are: ${items} - refer to logs.`)
```

Comment:
Going to check with Ed on this but we may not want an error message referencing logs to be client-facing and may want to just do the generic error message here. I'll check with him and get back with an update

Reply:
just for more information in making your decision - at the time we felt it might help the site owner and your team to debug the issue to have more info, but easy enough to pull this out and just leave a more generic error on the page.

Reply:
decision: remove public facing message that includes error-specific info